kindcat_custom.cue

```diff
@@ -10,6 +92,68 @@ package kindsys
 Custom: S={
 	_sharedKind
 
-	lineage: { name: S.machineName }
+	// pluginID is the unique identifier of the plugin which owns this Custom kind
+	pluginID: =~"^([a-z][a-z0-9-]*[a-z0-9])$"
+
+	// isCRD is true if the `crd` trait is present in the kind.
+	isCRD: S.crd != _|_
+
+	lineage: { 
+		name: S.machineName
+	}
 	lineageIsGroup: false
+
+	if isCRD {
+		// If the crd trait is defined, the schemas in the lineage must follow the format:
+		// {
+		//     "metadata": CommonMetadata & {...string}
+		//     "spec": {...}
+		//     "status": {...}
+		// }
+		lineage: joinSchema: _crdSchema
+	}
+
+	// crd contains properties specific to converting this kind to a Kubernetes CRD.
+	// Unlike in Core, crd is optional and is used as a signaling mechanism for whether the kind is intended to be registered as a Kubernetes CRD 
+	// and/or a resource in a compatible API server. When present, additional structure is enforced on the kind's lineage's schemas.
+	// When absent, a lineage's schema has no restrictions as it is assumed that a CRD or similar resource type will not be generated from it.
+	// 
+	// TODO: rather than `crd`, should this trait be something more generic, as it really indicates more if a resource should be available in a
+	// kubernetes-compatible APIServer, not specifically as CRD (though that _is_ an implementation)
+	crd?: {
+		// groupOverride is an override that is used in the crd trait if present.
+		// If left empty, plugin.id is used to generate the group name
+		groupOverride?: =~"^([a-z][a-z0-9-]{0,32}[a-z0-9])$"
+
+		// _computedGroups is a list of groups computed from information in the plugin trait.
+		// The first element is always the "most correct" one to use.
+		// This field could be inlined into `group`, but is separate for clarity.
+		_computedGroups: [
+			if S.crd.groupOverride != _|_ {
+				strings.ToLower(S.crd.groupOverride) + ".apps.grafana.com",
+			}
+			strings.ToLower(strings.Replace(S.pluginID, "-","_",-1)) + ".apps.grafana.com"
```

Comment:
As an open comment here as well: `<kind>.<pluginID/custom override>.apps.grafana.com`--do we want this to be `apps.grafana.com`, `ext.grafana.com`, `plugins.grafana.com`? I think it makes sense to group all custom kinds under one `<x>.grafana.com`, as that way they can't overlap with core kinds (which are `<kind>.grafana.com`), but what should <x> be?

CC: @bcotton @radiohead @mildwonkey

Reply:
`ext.grafana.com` works for me.

Side note – IMHO the core kinds should also be grouped under a subdomain, e.g. `core.grafana.com`.